src/main/java/jnr/unixsocket/UnixSocket.java

```diff
@@ -14,45 +14,147 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
+ * 
+ * ported from  https://github.com/softprops/unisockets/blob/master/unisockets-core/src/main/scala/Socket.scala 
  */
-
 package jnr.unixsocket;
 
-import jnr.constants.platform.SocketLevel;
-import jnr.constants.platform.SocketOption;
-import jnr.enxio.channels.NativeSocketChannel;
-import jnr.ffi.byref.IntByReference;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.SocketAddress;
+import java.nio.channels.Channels;
+import java.nio.channels.SocketChannel;
 
-import java.nio.ByteBuffer;
-import java.nio.ByteOrder;
-import java.nio.channels.Channel;
+class UnixSocket extends java.net.Socket {
 
-public class UnixSocket {
-    private final NativeSocketChannel channel;
+	private UnixSocketChannel chan;
```

Comment:
Shure. Since i also added static code analysis in the pom, I guess it does not hurt to add the checkstyle plugin in the pom as well. This way, in the future, we can let the build fail automatically when some source does not adhere to the coding-style... Going to be added in #35